app/src/main/java/com/vci/vectorcamapp/core/domain/cache/DeviceCache.kt

```diff
@@ -7,4 +7,7 @@ interface DeviceCache {
     suspend fun saveDevice(device: Device, programId: Int)
     suspend fun getDevice(): Device?
     suspend fun getProgramId(): Int?
+
+    fun getDeviceFlow(): Flow<Device?>
+    fun getProgramIdFlow(): Flow<Int?>
```

Comment:
Instead of getDeviceFlow and getProgramIdFlow, call these functions observeDevice, and observeProgramId